shared/teams/new-team/container.js

```diff
@@ -19,9 +19,15 @@ const mapDispatchToProps = (dispatch: Dispatch, {navigateUp}) => ({
   onBack: () => dispatch(navigateUp()),
 })
 
+const mergeProps = (stateProps, dispatchProps, ownProps) => ({
+  ...stateProps,
+  ...dispatchProps,
+  name: ownProps.routeProps.get('name'),
+})
+
 export default compose(
-  connect(mapStateToProps, mapDispatchToProps),
-  withState('name', 'onNameChange', ''),
+  connect(mapStateToProps, mapDispatchToProps, mergeProps),
+  withState('name', 'onNameChange', props => props.name),
```

Comment:
🐑 maybe `onNameChange` should throw away `teamCreationError`, so we can continue showing `headerText` when the user tries again

Reply:
Hmm, I can see arguments for both sides, e.g. a user might have entered a special character, and wants to not have to remember the entire error description before typing again if the error goes away.

I think once we display syntax errors immediately, that'll solve this also.